execution/function/operator.go

```diff
@@ -29,36 +30,56 @@ type functionOperator struct {
 
 	call         functionCall
 	scalarPoints [][]float64
+	model.OperatorTelemetry
+}
+
+func SetTelemetry(t model.OperatorTelemetry, opts *query.Options) {
+	t = &model.NoopTelemetry{}
+	if opts.EnableAnalysis {
+		t = &model.TrackedTelemetry{}
+	}
 }
 
 func NewFunctionOperator(funcExpr *parser.Call, nextOps []model.VectorOperator, stepsBatch int, opts *query.Options) (model.VectorOperator, error) {
 	// Some functions need to be handled in special operators
+
 	switch funcExpr.Func.Name {
 	case "scalar":
-		return &scalarFunctionOperator{
+		s := &scalarFunctionOperator{
 			next: nextOps[0],
 			pool: model.NewVectorPoolWithSize(stepsBatch, 1),
-		}, nil
+		}
+		SetTelemetry(s.OperatorTelemetry, opts)
+		return s, nil
+
 	case "label_join", "label_replace":
-		return &relabelFunctionOperator{
+		r := &relabelFunctionOperator{
 			next:     nextOps[0],
 			funcExpr: funcExpr,
-		}, nil
+		}
+		SetTelemetry(r.OperatorTelemetry, opts)
```

Comment:
With comment above we can then use it as,
```suggestion
			OperatorTelemetry: SetTelemetry(opts),
		}
```